integration-cli/docker_cli_daemon_test.go

```diff
@@ -798,7 +798,7 @@ func (s *DockerDaemonSuite) TestDaemonICCPing(c *testing.T) {
 
 	result := icmd.RunCommand("iptables", "-nvL", "FORWARD")
 	result.Assert(c, icmd.Success)
-	regex := fmt.Sprintf("DROP.*all.*%s.*%s", bridgeName, bridgeName)
+	regex := fmt.Sprintf("DROP.*0.*%s.*%s", bridgeName, bridgeName)
```

Comment:
I have almost no confidence in the beforehand regex (`DROP.*` is pretty loose) , now I have less :(
Are we sure this is correct? Ubuntu still has `all`.

If there really are differences here, we should handle this better as the test seems tied to the distro release.

Reply:
Yeah, not super-stoked with the test as well. Looking as well why on earth we used `-v` (verbose); it looks like without that, it's already a lot more parseable;

```bash
iptables -vL FORWARD
Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
pkts bytes target     prot opt in     out     source               destination
0     0    DOCKER-USER  all  --  any    any     anywhere             anywhere
0     0    DOCKER-ISOLATION-STAGE-1  all  --  any    any     anywhere             anywhere
0     0    ACCEPT     all  --  any    ext-bridge6  anywhere             anywhere             ctstate RELATED,ESTABLISHED
0     0    DOCKER     all  --  any    ext-bridge6  anywhere             anywhere
0     0    ACCEPT     all  --  ext-bridge6 !ext-bridge6  anywhere             anywhere
0     0    DROP       all  --  ext-bridge6 ext-bridge6  anywhere             anywhere
```

```bash
iptables -L FORWARD
Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
DOCKER-USER  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-1  all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
```

Reply:
🙈 nevermind; non-verbose excludes the `in` and `out` columns

Reply:
> Are we sure this is correct? Ubuntu still has all.

Yeah, iptables' `-n` means:

> Numeric  output.   IP  addresses  and port numbers will be printed in numeric format.

There was a bug in older iptables versions (bullseye is shipping iptables v1.8.7; and bookworm is shipping iptables v1.8.9) that made it not respect `-n`. This was fixed in https://git.netfilter.org/iptables/commit/?id=da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa.

But I agree this regex is not super great.

Reply:
- opened https://github.com/moby/moby/pull/46859 with an alternative